### ISSUE TITLE
Proxy + Consul Template

### ISSUE THREAD

**Issue by stevendborrelli** [CiscoCloud/microservices-infrastructure#42]
Provide a dynamic proxy based on consul-template & haproxy or nginx

**Comment by stevendborrelli** [comment]
@npateriya @keithchambers @dcheperd @BrianHicks 
Thoughts on storing shipped app data in consul so that we can update haproxy.

Consul stores every available service, but we only want the shipped proxy to expose shipped applications. What I am proposing is we update the consul K/V store and use those keys to query consul. 

For example, user creates app `test-app`, in the consul k/v store we'd store this as:

``shipped/group/user/apps/test-app/
``shipped/groups/user/apps/test-app/url``
``shipped/groups/user/apps/test-app/release_version``
``shipped/groups/users/apps/test-app/rollback_version``

When building an app we'd iterate over each of these keys and expose the release version via the port. 

```{{range tree "groups@dc1"}}
.. iterate over each user

create haproxy server links:
       {{ range version.app@dc1 }}
      server {{ .Address }}:{{.Port}}

{{end}}```

**Comment by stevendborrelli** [comment]
Merged into 0.2-dev.
